src/components/WeatherDisplay.jsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import "./WeatherDisplay.css";
+
+const WeatherDisplay = ({
+  currentWeather,
+  selectedFormat,
+  convertToCelsius
+}) => {
+  if (!(currentWeather && currentWeather.main && currentWeather.weather)) {
+    return <div>Loading...</div>;
+  }
+
+  const iconSrc = `http://openweathermap.org/img/w/${currentWeather.weather[0].icon}.png`;
+
+  return (
+    <div className="ui row">
+      <div className="left floated left aligned six wide column">
+        <h1>
+          Current Temp:{" "}
+          {selectedFormat === "F"
+            ? Math.round(currentWeather.main.temp)
+            : convertToCelsius(Math.round(currentWeather.main.temp))}{" "}
+          °
+        </h1>
+        <img
+          className="current-weather-icon"
+          src={iconSrc}
+          alt={currentWeather.weather[0].main}
+        />
+        <p>{currentWeather.weather[0].main}</p>
+        <p>
+          Feels Like:{" "}
+          {selectedFormat === "F"
```

Comment:
I notice there is this repeated logic/check a few places in these components. Do you have any thoughts on how it may be more reusable?

Reply:
Definitely. I handled this task as the last piece of the implementation with limited time, so that's the reason for the repeated logic, as I was just trying to get it working. There are of course many ways we could refactor this, but one way would be to create a helper function that takes in a "weather" object and a format, and converts the relevant temperatures (the ones the UI is using) to that format. We would then update the state with this new object as "currentWeather". This would mean that we don't have to check for the temperature format anywhere in this component, as the object being passed into this component would already have the correct units. Additionally, when rendering the temperatures in the UI like "34° F" we wouldn't need to do any conditional checks, we would just render the `selectedFormat` prop in all locations where we want to show the format, like "F" or "C".